File: command-pattern-master/command-pattern-master/java/src/robotcontroller/invoker/Button.java
package robotcontroller.invoker;

import robotcontroller.commands.ICommand;

/**
 * Created by ramiz on 11/15/17.
 */
public class Button implements IButton {
    private ICommand command;

    public Button(ICommand command) {
        this.command = command;
    }

    public void setCommand(ICommand command) {
        this.command = command;
    }

    public ICommand getCommand() {
        return command;
    }

    @Override
    public void press() {
        command.execute();
    }
}

File: command-pattern-master/command-pattern-master/c++/remotecontroller/button.cpp
/*
 * button.cpp
 *
 *  Created on: Nov 16, 2017
 *      Author: ramiz
 */

#include "button.h"

Button::Button(ICommand* command) {
  this->command_ = command;
}

Button::~Button() {

}

void Button::Press() {
  this->command_->Execute();
}

void Button::SetCommand(ICommand * command) {
  this->command_ = command;
}

ICommand* Button::GetCommand() {
  return this->command_;
}

